### PR TITLE
improve scantree.py by adding --ignore option

### PR DESCRIPTION
**Pull request by skeept**

### CODE REVIEW

File: doc/denite.txt [hunk 1]
```diff
@@ -88,8 +88,6 @@ INSTALL							*denite-install*
 Note: denite requires Neovim(the latest is recommended) or Vim8.0+ with
 |:python3| enabled.
 
-Note: You need to install Python3.5+ or `pip3 install typing`.
```
Comment (Shougo) [inline doc/denite.txt]: Please don't remove it.

**Comment by Shougo** [comment]
Please fix the CI errors.

File: rplugin/python3/denite/source/file_rec.py [hunk 2]
```diff
@@ -9,6 +9,29 @@
...
+def parse_command_for_scantree(cmd):
+    """Given the user choice for --ignore get the corresponding value"""
+
+    parser = argparse.ArgumentParser(description="parse scantree options")
+    parser.add_argument('--ignore', type=str, default='use_wildignore')
+
+    # the first name on the list is 'scantree.py'
+    args = parser.parse_args(cmd[1:])
+    if args.ignore == 'use_wildignore':
+        # how to get the value of vim variable?
```
Comment (skeept) [inline rplugin/python3/denite/source/file_rec.py]: @Shougo can you help here?
How to get the value of the option wildignore from vim.

I think this is possible but I don't know how to do it.

File: rplugin/python3/denite/source/file_rec.py [hunk 3]
```diff
@@ -6,32 +6,30 @@
 
 from .base import Base
 from denite.process import Process
-from os import path, pardir
+from os import path
 from os.path import relpath, isabs, isdir, join
 from denite.util import parse_command, abspath
 import argparse
 import os
+import vim
```
Comment (skeept) [inline rplugin/python3/denite/source/file_rec.py]: This works for me when I run it from inside vim but it fails the CI checks.
Can you advice here how to fix this?

@Shougo

Comment (Shougo) [reply rplugin/python3/denite/source/file_rec.py]: Please don't use this.
Please use `self.vim` instead.

File: rplugin/python3/denite/source/file_rec.py (hunk 2 above)
Comment (Shougo) [reply rplugin/python3/denite/source/file_rec.py]: `self.vim.options['wildignore']`

**Comment by Shougo** [comment]
Merged.

**Comment by skeept** [comment]
@Shougo  Thank you for merging. 

I was going to ask you if you had a chance of testing in your enviroment. I just use windows but I think it should work in other environments as well.

**Comment by Shougo** [comment]
> I was going to ask you if you had a chance of testing in your enviroment. I just use windows but I think it should work in other environments as well.

I have tested it today.
The code is buggy.  I will fix it later.

**Comment by Shougo** [comment]
I have fixed the errors.

**Comment by skeept** [comment]
Thank you for the fixing the errors.. I looked at the code they are good.

**Comment by Shougo** [comment]
:-)

**Comment by Shougo** [comment]
> I was going to ask you if you had a chance of testing in your enviroment. I just use windows but I think it should work in other environments as well.

It works on Linux environment.

**Comment by skeept** [comment]
It is necessary another small fix because python 3 doesn't always work in windows. Usually the python windows executable is python.exe. I added a small branch to fix that. Please review and merge when you have a chance.

**Pull request merged**